src/statement-list/statement_list_parser.cpp

```diff
@@ -20,234 +21,467 @@ statement_list_parsert statement_list_parser;
-  parsert::clear();
-  parse_tree.clear();
-}
-
-void statement_list_parsert::print_tree(std::ostream &out) const
+int yystatement_listerror(const std::string &error)
 {
-  parse_tree.output(out);
+  statement_list_parser.parse_error(error, yystatement_listtext);
+  return 0;
 }
 
-irep_idt statement_list_parsert::find_name(const exprt &root)
+/// Searches for the name of the TIA module inside of its root
+/// expression.
+/// \param root: Expression that includes the element's name as a
+///   direct operand.
+/// \return The name of the function element.
+irep_idt find_name(const exprt &root)
 {
-  for(exprt op : root.operands())
+  for(const exprt &op : root.operands())
   {
     if(op.get(ID_statement_list_type) == ID_statement_list_identifier)
       return op.get(ID_value);
   }
-
   UNREACHABLE; // Root expression should always have a name
 }
 
-float statement_list_parsert::find_version(const exprt &root)
+/// Searches for the version of the TIA module inside of its root
+/// expression.
+/// \param root: Expression that includes the element's version as a
+///   direct operand.
+/// \return The version of the element.
+std::string find_version(const exprt &root)
 {
-  for(exprt op : root.operands())
+  for(const exprt &op : root.operands())
   {
     if(op.type().id() == ID_statement_list_version)
-      return std::stof(op.get(ID_value).c_str());
+      return id2string(op.get(ID_value));
   }
-
   UNREACHABLE; // Root expression should always have a version
 }
 
-exprt statement_list_parsert::find_variable_list(const exprt &root)
+/// Searches for the return type of a function inside of its root
+/// expression.
+/// \param root: Expression that includes the function's return type as a
+///   direct operand.
+/// \return The return value of the function.
+typet find_return_value(const exprt &root)
 {
-  for(exprt op : root.operands())
+  INVARIANT(
+    root.id() == ID_statement_list_function,
+    "Expression ID should be statement_list_function");
+
+  for(const exprt &op : root.operands())
+  {
+    if(op.get(ID_statement_list_type) == ID_statement_list_return)
+      return typet(op.id());
+  }
+
+  UNREACHABLE; // Root expression of FC should always have a return value
+}
+
+/// Searches for the variable list of the TIA module inside of its root
+/// expression.
+/// \param root: Expression that includes the element's variable list as a
+///   direct operand.
+/// \return The variable list of the element.
+exprt find_variable_list(const exprt &root)
+{
+  for(const exprt &op : root.operands())
   {
     if(op.id() == ID_statement_list_var_decls)
       return op;
   }
   UNREACHABLE; // Root expression should always have a variable list
 }
 
-exprt statement_list_parsert::find_network_list(const exprt &root)
+/// Adds all input variable declarations to the given element.
+/// \param module: The TIA element to which the variables belong.
+/// \param input_vars: The root expression of a input variable list.
+void fill_input_vars(
```

Comment:
`static` if it is only used from within the compilation unit (several occurrences)